packages/network-controller/tests/NetworkController.test.ts

```diff
@@ -2686,7 +2686,198 @@ describe('NetworkController', () => {
+                  {
+                    request: {
+                      method: 'eth_getBlockByNumber',
+                    },
+                    response: {
+                      result: POST_1559_BLOCK,
+                    },
+                    beforeCompleting: async () => {
+                      messenger.clearSubscriptions();
+                      await controller.destroy();
+                    },
+                  },
+                ]),
+              ];
+              const fakeNetworkClients = [buildFakeClient(fakeProviders[0])];
+              mockCreateNetworkClient()
+                .calledWith({
+                  chainId: toHex(1337),
+                  rpcUrl: 'https://mock-rpc-url',
+                  type: NetworkClientType.Custom,
+                  ticker: 'TEST',
+                })
+                .mockReturnValue(fakeNetworkClients[0]);
+              await controller.initializeProvider();
+
+              expect(
+                Object.keys(
+                  controller.state.networksMetadata['https://mock-rpc-url']
+                    .EIPS,
+                ).length,
+              ).toStrictEqual(0);
+            },
+          );
+        });
+
+        it('does not emit infuraIsUnblocked', async () => {
+          await withController(
+            {
+              state: {
+                providerConfig: buildProviderConfig({
+                  type: NetworkType.rpc,
+                  chainId: toHex(1337),
+                  rpcUrl: 'https://mock-rpc-url',
+                }),
+              },
+              infuraProjectId: 'some-infura-project-id',
+            },
+            async ({ controller, messenger }) => {
+              const fakeProviders = [
+                buildFakeProvider([
+                  // Called during provider initialization
+                  {
+                    request: {
+                      method: 'eth_getBlockByNumber',
+                    },
+                    response: SUCCESSFUL_ETH_GET_BLOCK_BY_NUMBER_RESPONSE,
+                    beforeCompleting: async () => {
+                      messenger.clearSubscriptions();
+                      await controller.destroy();
+                    },
+                  },
+                ]),
+              ];
+              const fakeNetworkClients = [buildFakeClient(fakeProviders[0])];
+              mockCreateNetworkClient()
+                .calledWith({
+                  chainId: toHex(1337),
+                  rpcUrl: 'https://mock-rpc-url',
+                  type: NetworkClientType.Custom,
+                  ticker: 'TEST',
+                })
+                .mockReturnValue(fakeNetworkClients[0]);
+
+              await controller.initializeProvider();
+
+              const promiseForNoInfuraIsUnblockedEvents =
+                waitForPublishedEvents({
+                  messenger,
+                  eventType: 'NetworkController:infuraIsBlocked',
+                  count: 0,
```

Comment:
We should ensure that we are selecting an Infura network here (because that's when `infuraIsUnblocked` would ordinary get emitted). We should also listen for `infuraIsUnblocked` (not `infuraIsBlocked`) to not get emitted:

``` suggestion
              state: {
	              providerConfig: buildProviderConfig({
	                type: NetworkType.goerli,
	              }),
	            },
              infuraProjectId: 'some-infura-project-id',
            },
            async ({ controller, messenger }) => {
              const fakeProviders = [
                buildFakeProvider([
                  // Called during provider initialization
                  {
                    request: {
                      method: 'eth_getBlockByNumber',
                    },
                    response: SUCCESSFUL_ETH_GET_BLOCK_BY_NUMBER_RESPONSE,
                    beforeCompleting: async () => {
                      messenger.clearSubscriptions();
                      await controller.destroy();
                    },
                  },
                ]),
              ];
              const fakeNetworkClients = [buildFakeClient(fakeProviders[0])];
              mockCreateNetworkClient()
                .calledWith({
                  network: NetworkType.goerli,
	                infuraProjectId: 'some-infura-project-id',
	                chainId: BUILT_IN_NETWORKS[NetworkType.goerli].chainId,
	                ticker: BUILT_IN_NETWORKS[NetworkType.goerli].ticker,
                })
                .mockReturnValue(fakeNetworkClients[0]);

              await controller.initializeProvider();

              const promiseForNoInfuraIsUnblockedEvents =
                waitForPublishedEvents({
                  messenger,
                  eventType: 'NetworkController:infuraIsUnblocked',